packages/runner/lib/exec.ts

```diff
@@ -13,20 +13,17 @@ import { logger } from './utils.js';
 
 export async function exec(
     nangoProps: NangoProps,
-    isInvokedImmediately: boolean,
-    isWebhook: boolean,
+    scriptType: 'sync' | 'action' | 'webhook' | 'post-connection-script',
```

Comment:
currently in run.service we have some `isAction`, `isWebhook` booleans that are transformed into `isInvokedImmediately` and again in `isAction` in this file.
In order to be consistent and make it easy to follow, I am gonna change those boolean into a union type that is passed along the chain from jobs to runner and back to jobs